MobileWallet/Screens/Profile/ProfileViewController.swift

```diff
@@ -81,54 +73,34 @@ class ProfileViewController: UIViewController {
 
     // MARK: - Private functions
 
-    private func setupCloseButton() {
-        closeButton.setImage(Theme.shared.images.close!, for: .normal)
-        closeButton.addTarget(self, action: #selector(onDismissAction), for: .touchUpInside)
-        view.addSubview(closeButton)
-        closeButton.translatesAutoresizingMaskIntoConstraints = false
+    private func setupNavigationBar() {
+        navigationBar.title = NSLocalizedString("My Profile", comment: "ProfileViewController title")
+        navigationBar.backgroundColor =  Theme.shared.colors.profileBackground
 
-        closeButton.widthAnchor.constraint(equalToConstant: 40).isActive = true
-        closeButton.heightAnchor.constraint(equalToConstant: 40).isActive = true
-        closeButton.leadingAnchor.constraint(equalTo: view.leadingAnchor, constant: 25).isActive = true
-        closeButton.topAnchor.constraint(equalTo: view.layoutMarginsGuide.topAnchor, constant: 20).isActive = true
-    }
+        navigationBar.rightButtonAction = { [weak self] in
+            let navigationController = AlwaysPoppableNavigationController(rootViewController: SettingsViewController())
+            navigationController.styleNavigatorBar(isHidden: true)
+            self?.present(navigationController, animated: true, completion: nil)
```

Comment:
Not sure about this presentation style, `pushViewController` may be better, but we should ask the product team after the initial release.

Reply:
we should present this controller as modal. https://zpl.io/advgkK7